CI: Test Flake on test_random.test_covariance_game

I first observed this on CI on Python 3.7 on Windows. Turns out it has happened on Ubuntu as well: https://github.com/QuantEcon/QuantEcon.py/pull/566#issuecomment-757078096.
The log:
```
FAIL: test_random.test_covariance_game
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/runner/work/QuantEcon.py/QuantEcon.py/quantecon/game_theory/tests/test_random.py", line 33, in test_covariance_game
    assert_allclose(payoff_profile[i], payoff_profile[-1], atol=1e-8)
  File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/numpy/testing/_private/utils.py", line 1528, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/numpy/testing/_private/utils.py", line 840, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=1e-08
```

--- a/quantecon/game_theory/tests/test_random.py
+++ b/quantecon/game_theory/tests/test_random.py
@@ -28,8 +28,8 @@ def test_covariance_game():
     rho = 1
     g = covariance_game(nums_actions, rho=rho)
     for a in np.ndindex(*nums_actions):
+        payoff_profile = g.payoff_profile_array[a]
         for i in range(N-1):
-            payoff_profile = g.payoff_profile_array[a]
             assert_allclose(payoff_profile[i], payoff_profile[-1], atol=1e-7)
 
     rho = -1 / (N - 1)

--- a/quantecon/game_theory/tests/test_random.py
+++ b/quantecon/game_theory/tests/test_random.py
@@ -30,7 +30,7 @@ def test_covariance_game():
     for a in np.ndindex(*nums_actions):
         for i in range(N-1):
             payoff_profile = g.payoff_profile_array[a]
-            assert_allclose(payoff_profile[i], payoff_profile[-1], atol=1e-8)
+            assert_allclose(payoff_profile[i], payoff_profile[-1], atol=1e-7)
 
     rho = -1 / (N - 1)
     g = covariance_game(nums_actions, rho=rho)

